uppsrc/CtrlCore/CtrlCore.h

```diff
@@ -685,6 +685,7 @@ class Ctrl : public Pte<Ctrl> {
 	void    SyncScroll();
 	void    Refresh0(const Rect& area);
 	void    PaintCaret(SystemDraw& w);
+	int16        caretx, carety, caretcx, caretcy;//8  //aris002
```

Comment:
No need for introducing new variable there is:
`static Rect      caretRect;`

As I wrote please reference to this commit https://github.com/ultimatepp/ultimatepp/commit/c7c4d97499be4b3f4a6423103188cabe2ff3e570

Reply:
1. As we spoke on Discord and you said "The first commit on the list should be related to changes that broke pure X11 compilation." This exact line had been used in Ctrl structure before and now it exactly restores the compilation ability, gives a chance to debug and investigate possible harmonisation with other caret related changes.  2. Now I will try to trace those changes. 3. Also, this is exactly the 'strategic' problem I said I wanted to discuss with CtrlCore chief architects before committing this code.  4. On the other hand, that is why I proposed to create a separate X11 dev  branch on ultimatepp/ultimatepp for more developers to participate. And I wanted to make live easier/faster by creating a compile/debug-able branch. 5. But I am open to upp or your style of work.